src/utils.py

```diff
@@ -56,26 +58,32 @@ def allocate_array_space(
     last_container = ensure_parent_containers(seg_tiled_uri, seg_tiled_api_key)
 
     print(f"@@@@@@@@@@    last_container   {last_container.uri}")
-    assert uid not in last_container.keys(), f'uid_save: {uid} already existed in Tiled Server'
-
+    assert (
+        uid not in last_container.keys()
+    ), f"uid_save: {uid} already existed in Tiled Server"
 
     last_container = last_container.create_container(key=uid)
+
     array_shape = (
         tiled_dataset.mask_client.shape
         if tiled_dataset.mask_client
         else tiled_dataset.data_client.shape
     )
+    # In case we have 2d array for the single mask case, where the ArrayClient will return a 2d array.
+    if len(array_shape) == 2:
```

Comment:
Did you encounter this case when writing a single mask frame through the dash interface? If yes, I should investigate that code again.

Reply:
No, we manually ingested the annotations in tiled for testing yesterday. I accidentally wrote the mask as a 2D array since it was just 1 frame. If the dash interface saves the masks as a 3D array even when there is a single frame, we can remove these lines of code.

Reply:
Yes, in one of our testing case yesterday, @taxe10 manually ingested some data into the local Tiled Server, and there was only 1 mask for the `mask_client`. We ended up with `dataset.mask_client` resulted in a shape of 2d array structure, and ran into "index of out bound" issues when retrieving the block information. So this is a layer added to mitigate the issue.

Please keep an eye to see whether this is also the case for the current pipeline, if we ingest a single mask from the dash app.

Reply:
Makes sense. This is not the result when we are exporting the mask from the interface.